pytest_bdd/scenario.py

```diff
@@ -59,10 +58,15 @@ def find_argumented_step_fixture_name(name, type_, fixturemanager, request=None)
     # happens to be that _arg2fixturedefs is changed during the iteration so we use a copy
     for fixturename, fixturedefs in list(fixturemanager._arg2fixturedefs.items()):
         for fixturedef in fixturedefs:
-            parser = getattr(fixturedef.func, "parser", None)
+            if fixturedef.func.__name__ == 'lazy_step_func':
```

Comment:
what are we fixing here exactly?

Reply:
It is a good question ;)

It fixes https://github.com/pytest-dev/pytest-bdd/issues/250. For some unknown reason on the latest pytest above `fixturedef.func` don't have `parser` attribute -> even that on `pytest==3.6.4` `parser` attribute is there.

Because of that difference, `pytest-bdd` was not able to find some fixtures and tests were failing.

My guess is that previous `pytest` is calling `fixturedef.func` before our `find_argumented_step_fixture_name` is called.

The solution I found is to call the `func` if it is `lazy_step_func`. That call assigns  `parser` attribute and all work as before.

I found that solution during debugging but I don't know the root cause of the issue.

Reply:
I would leave this change out and make a second PR for that, to be honest. It would make this case easier to solve.

Reply:
As you suggested I have moved it to separate pull request.

The only problem is that without that fix tests start to fail. May need to merge fix locally to this branch to make sure that all is good.